charts/redpanda/templates/post-install-upgrade-job.yaml

```diff
@@ -76,23 +76,27 @@ spec:
 {{- if .Values.auth.sasl.enabled }}
   {{- $values := .Values }}
   {{- range $user := .Values.auth.sasl.users }}
-            # To avoid `set -e` from exiting the command when a user exists; catch the stderr output and exit codes into `creation_result`
-            # and `creation_result_exit_code` for use later
-            creation_result=$(rpk acl user create {{ $user.name }} -p {{ $user.password | quote }} --mechanism {{ include "sasl-user-mechanism" (dict "user" $user "Values" $values) }} {{ template "rpk-common-flags" $ }} 2>&1) && creation_result_exit_code=$? || creation_result_exit_code=$?
-
-            # On a non-success exit code
+    {{- if $user.password }}
+            PASSWORD={{ $user.password | quote }}
```

Comment:
Should we even allow that? In my understanding any Password should be protected and not given as a plain text of the job resource.

Reply:
That was the previous behavior? I can change this to just push any password into a secret, but that would be a larger PR and probably one that addresses this directly.

Reply:
@RafalKorepta is right, we need to protect users credentials. As proposed by @alejandroEsc the right way to do it is to push it into etcd(vault, etc) as a secret and use it from there.

Reply:
That would be a breaking change. We can issue a warning, but not break existing users.